command/agent/alloc_endpoint_test.go

```diff
@@ -394,6 +394,32 @@ func TestHTTP_AllocRestart_ACL(t *testing.T) {
 	})
 }
 
+func TestHTTP_AllocStop(t *testing.T) {
+	t.Parallel()
+	httpTest(t, nil, func(s *TestAgent) {
+		// Directly manipulate the state
+		state := s.Agent.server.State()
+		alloc := mock.Alloc()
+		require := require.New(t)
+		require.NoError(state.UpsertJobSummary(999, mock.JobSummary(alloc.JobID)))
+
+		require.NoError(state.UpsertAllocs(1000, []*structs.Allocation{alloc}))
+
+		// Make the HTTP request
+		req, err := http.NewRequest("POST", "/v1/allocation/"+alloc.ID+"/stop", nil)
```

Comment:
Maybe throw a request for a bogus alloc id at the bottom and assert 404.

Reply:
+1